src/librustc_builtin_macros/source_util.rs

```diff
@@ -110,6 +112,15 @@ pub fn expand_include<'cx>(
     };
     let p = new_sub_parser_from_file(cx.parse_sess(), &file, None, sp);
 
+    // If in the included file we have e.g., `mod bar;`,
+    // then the path of `bar.rs` should be relative to the directory of `file`.
+    // See https://github.com/rust-lang/rust/pull/69838/files#r395217057 for a discussion.
+    // `MacroExpander::fully_expand_fragment` later restores, so "stack discipline" is maintained.
```

Comment:
This claim is a bit subtle, but this is what I inferred when reading the code in `fully_expand_fragment`. If this is not correct, then the alternative would be to add a flag to `SyntaxExtension` or something and deal with it in `rustc_expand::expand`.